duckreg/estimators.py

```diff
@@ -54,15 +64,57 @@ def compress_data(self):
         )
         self.df_compressed.eval(create_means, inplace=True)
 
+    def collect_data(self, data: pd.DataFrame) -> pd.DataFrame:
+
+        y = data[f"mean_{self.outcome_vars[0]}"].values
+        X = data[self.covars].values
+        n = data["count"].values
+
+        # y, X, w need to be two-dimensional for the demean function
+        y = y.reshape(-1, 1) if y.ndim == 1 else y
+        X = X.reshape(-1, 1) if X.ndim == 1 else X
+
+        if self.fevars:
+            # fe needs to contain of only integers for
+            # the demean function to work
+            fe = _convert_to_int(data[self.fevars])
+            fe = fe.reshape(-1, 1) if fe.ndim == 1 else fe
+
+            y, _ = demean(x=y, flist=fe, weights=n)
```

Comment:
this looks reasonable and makes sense in my head, although i wasn't able to find examples where demeaning post compression made much of a dent on runtime.

Reply:
I think you're right! I think this might only make a difference if the fixed effects are of relatively high cardinality, and as a result, the compressed data set is relatively "long"? Should I take this out, then?

Reply:
i don't see any harm in keeping it; this only gets triggered conditional on the presence of `|` in the formula anyway